granulate_utils/linux/elf.py

```diff
@@ -90,3 +97,52 @@ def is_statically_linked(path: str) -> bool:
             if segment.header.p_type == "PT_DYNAMIC":
                 return False
     return True
+
+
+def get_symbol_addr(path: str, sym_name: str) -> Optional[int]:
+    with open(path, "rb") as f:
+        elf = ELFFile(f)
+        symtab = elf.get_section_by_name(".symtab")
+        if symtab is None:
+            return None
+        symbols = symtab.get_symbol_by_name(sym_name)
+        if symbols is None or len(symbols) != 1:
+            return None
+        return symbols[0].entry.st_value
+
+
+def get_dt_needed(path: str) -> List[str]:
+    dt_needed: List[str] = []
+    with open(path, "rb") as f:
+        elf = ELFFile(f)
+        dynamic_section = elf.get_section_by_name(".dynamic")
+        if dynamic_section is None:
+            return dt_needed
+        for tag in dynamic_section.iter_tags():
+            if tag.entry.d_tag == "DT_NEEDED":
+                dt_needed.append(tag.needed)
+    return dt_needed
```

Comment:
```suggestion
        return [tag.needed for tag in dynamic_section.iter_tags() if tag.entry.d_tag == "DT_NEEDED"]
```

Reply:
Also changed to return `None` if no dynamic section instead of empty list